packages/sdk/src/universe/builders/single-token/index.ts

```diff
@@ -0,0 +1,153 @@
+        tokenId: nftInfo.assetType.tokenId!,
+        price: paymentInfo.value,
+        paymentToken:
+          paymentInfo.assetType.assetClass === AssetClass.ETH
+            ? constants.AddressZero
+            : lc(paymentInfo.assetType.contract!),
+        salt: n(order.params.salt),
+        startTime: order.params.start,
+        endTime: order.params.end,
+        signature: order.params.signature,
+        tokenAmount: n(nftInfo.value),
+        revenueSplits: order.params.data.revenueSplits || [],
+      });
+
+      if (!copyOrder) {
+        return false;
+      }
+
+      if (copyOrder.hashOrderKey() !== order.hashOrderKey()) {
+        return false;
+      }
+    } catch {
+      return false;
+    }
+
+    return true;
+  }
+
+  public build(params: Types.BaseBuildParams) {
+    this.defaultInitialize(params);
+    const nftInfo = {
+      assetType: {
+        assetClass: params.tokenKind.toUpperCase(),
+        contract: lc(params.contract),
+        tokenId: params.tokenId,
+      },
+      value: s(params.tokenAmount || 1),
+    };
+
+    const paymentInfo = {
+      assetType: {
+        ...(params.paymentToken && params.paymentToken !== constants.AddressZero
+          ? {
+              assetClass: AssetClass.ERC20,
+              contract: lc(params.paymentToken),
+            }
+          : {
+              assetClass: AssetClass.ETH,
+            }),
+      },
+      value: params.price,
+    };
+    return new Order(this.chainId, {
+      kind: "single-token",
+      //TODO: Extract this to a constant
+      type: "UNIVERSE_V1",
+      side: params.side === "buy" ? 0 : 1,
+      maker: params.maker,
+      make: params.side === "buy" ? paymentInfo : nftInfo,
+      taker: constants.AddressZero,
+      take: params.side === "buy" ? nftInfo : paymentInfo,
+      salt: s(params.salt),
+      start: params.startTime,
+      end: params.endTime!,
+      data: {
+        dataType: params.revenueSplits?.length
+          ? Constants.ORDER_DATA
+          : Constants.DATA_TYPE_0X,
+        revenueSplits: params.revenueSplits || [],
+      },
+      signature: params?.signature,
+    });
+  }
+
+  public buildMatching(
+    order: Types.Order,
+    taker: string,
+    data: { amount?: string }
+  ) {
+    const rightOrder = {
+      type: order.type,
+      maker: taker,
+      taker: constants.AddressZero,
+      make: JSON.parse(JSON.stringify(order.take)),
+      take: JSON.parse(JSON.stringify(order.make)),
+      salt: 0,
+      start: order.start,
+      end: order.end,
+      data: {
+        dataType: order.data.revenueSplits?.length
+          ? Constants.ORDER_DATA
+          : Constants.DATA_TYPE_0X,
+        revenueSplits: order.data.revenueSplits || [],
+      },
+    };
+
+    // for erc1155 we need to take the value from request (the amount parameter)
+    if (AssetClass.ERC1155 == order.make.assetType.assetClass) {
+      const availableAmount = Number(order.make.value) - Number(order.fill);
```

Comment:
The available fill amount check will be done in the indexer, we can exclude it from the SDK.

Reply:
I've removed it but I'm not sure where exactly to put in the indexer

Reply:
It's the quantity_remaining field on the orders table:
https://github.com/reservoirprotocol/indexer/blob/a7265695193b75d3fe22cb8b09f7d27a6de5b6dc/src/orderbook/orders/seaport/index.ts#L407

That gets deducted when we ingest fill events:
https://github.com/reservoirprotocol/indexer/blob/a7265695193b75d3fe22cb8b09f7d27a6de5b6dc/src/sync/events/storage/fill-events/partial.ts#L133-L134